mix/QVariableDefinition.cpp

```diff
@@ -53,3 +55,95 @@ QVariableDefinition* QVariableDefinitionList::val(int _idx)
 		return nullptr;
 	return m_def.at(_idx);
 }
+
+/*
+ * QIntType
+ */
+dev::bytes QIntType::encodeValue()
+{
+	dev::bigint i(value().toStdString());
+	if (i < 0)
+		i = i + dev::bigint("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") + 1;
+	std::ostringstream s;
+	s << std::hex << "0x" << i;
+	return padded(jsToBytes(s.str()), 32);
+}
+
+void QIntType::decodeValue(std::string const& _rawValue)
+{
+	std::string rawParam = _rawValue.substr(0, 32 * 2);
+	dev::bigint bigint = dev::bigint("0x" + rawParam);
+	if (((bigint >> 32) & 1) == 1)
+		bigint = bigint - dev::bigint("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff") - 1;
+	std::ostringstream s;
+	s << std::dec << bigint;
+	setValue(QString::fromStdString(s.str()));
+}
+
+/*
+ * QHashType
+ */
+dev::bytes QHashType::encodeValue()
+{
+	return padded(jsToBytes("0x" + value().toStdString()), 32);
+}
+
+void QHashType::decodeValue(std::string const& _rawValue)
+{
+	std::string rawParam = _rawValue.substr(0, 32 * 2);
+	std::string unPadded = unpadLeft(rawParam);
+	setValue(QString::fromStdString(unPadded));
+}
+
+/*
+ * QRealType
+ */
+dev::bytes QRealType::encodeValue()
+{
+	return bytes();
+}
+
+void QRealType::decodeValue(std::string const& _rawValue)
+{
+	Q_UNUSED(_rawValue);
+}
+
+/*
+ * QStringType
+ */
+dev::bytes QStringType::encodeValue()
+{
+	qDebug() << QString::fromStdString(toJS(paddedRight(asBytes(value().toStdString()), 32)));
+	return paddedRight(asBytes(value().toStdString()), 32);
+}
+
+void QStringType::decodeValue(std::string const& _rawValue)
+{
+	std::string rawParam = _rawValue.substr(0, 32 * 2);
+	rawParam = unpadRight(rawParam);
```

Comment:
From the Wiki, we need to right aligned the string: 
string<N>: binary type of N bytes, N > 0. Assumed to be UTF-8 encoded, right-aligned, zero-passed data.